ledger-core/virtual/integration/mock/typepublishchecker.go

```diff
@@ -634,3 +671,19 @@ func (p *TypePublishChecker) MinimockWait(timeout mm_time.Duration) {
 		}
 	}
 }
+
+func waitCounterIndefinitely(counter *atomickit.Int, count int) synckit.SignalChannel {
```

Comment:
There should be mechanism (even not "thread safe") to stop waiting, in case we stop test with timeout.

Reply:
I couldn't find good solution to implement such mechanizm and decided that it's ok. Do you have simple and straightforward solution?

Reply:
do we really expect test failures to cleanup everything after?